src/core/DataFile.cpp

```diff
@@ -264,24 +276,89 @@ void DataFile::write( QTextStream & _strm )
 
 
 
-bool DataFile::writeFile( const QString& filename )
+bool DataFile::writeFile(const QString& filename, bool withResources)
 {
-	const QString fullName = nameWithExtension( filename );
+	// Small lambda function for displaying errors
+	auto showError = [this](QString title, QString body){
+		if (gui)
+		{
+			QMessageBox mb;
+			mb.setWindowTitle(title);
+			mb.setText(body);
+			mb.setIcon(QMessageBox::Warning);
+			mb.setStandardButtons(QMessageBox::Ok);
+			mb.exec();
+		}
+		else
+		{
+			qWarning() << body;
+		}
+	};
+
+	// If we are saving without resources, filename is just the file we are
+	// saving to. If we are saving with resources (project bundle), filename
+	// will be used (discarding extensions) to create a folder where the
+	// bundle will be saved in
+
+	QFileInfo fInfo(filename);
+
+	const QString bundleDir = fInfo.path() + "/" + fInfo.fileName().section('.', 0, 0);
+	const QString resourcesDir = bundleDir + "/resources";
+	const QString fullName = withResources
+		? nameWithExtension(bundleDir + "/" + fInfo.fileName())
+		: nameWithExtension(filename);
 	const QString fullNameTemp = fullName + ".new";
 	const QString fullNameBak = fullName + ".bak";
 
-	QFile outfile( fullNameTemp );
-
-	if( !outfile.open( QIODevice::WriteOnly | QIODevice::Truncate ) )
+	// If we are saving with resources, setup the bundle folder first
+	if (withResources)
 	{
-		if( gui )
+		// First check if there's a bundle folder with the same name in
+		// the path already. If so, warns user that we can't overwrite a
+		// project bundle.
+		if (QDir(bundleDir).exists())
 		{
-			QMessageBox::critical( NULL,
-				SongEditor::tr( "Could not write file" ),
-				SongEditor::tr( "Could not open %1 for writing. You probably are not permitted to "
-								"write to this file. Please make sure you have write-access to "
-								"the file and try again." ).arg( fullName ) );
+			showError(SongEditor::tr("Operation denied"),
```

Comment:
Oh, forgot this comment. I'm not sure, think the difference is a bit subjective in this case. I mean, the program doesn't crash because the project didn't save, it just stays in the project you had before, so I can see a point to using a Warning instead of the Critical we had before. The difference is basically stylistic, maybe the critical message box has a different parent too if I understood right, but that's pretty much it apparently. Is it even worth changing?

Reply:
No, just wanted to make sure you saw the difference. Feel free to resolve the conversation.